src/Visualization/Python/PlotSizeControl.py

```diff
@@ -0,0 +1,361 @@
+            )
+            damping_timescale = np.concatenate(
+                [damping_timescale, diagnostics_data[:, 22]]
+            )
+
+    # Apply stylesheet
+    if stylesheet is not None:
+        plt.style.use(stylesheet)
+
+    # Set up plots
+    fig, axes = plt.subplots(7, 1, sharex=True)
+    plt.rcParams["text.usetex"] = True
+
+    # Plot the selected quantities. Groupings are based both on relationship
+    # of quantities and also relative scales of quantities
+    axes = list(axes)
+    # Damping times are sort of the odd ones out. Put it with control error but
+    # on a different y-axis
+    top1 = axes[0].plot(times, control_error, label="Control Error")
+    axes.append(axes[0].twinx())
+    top2 = axes[-1].plot(
+        times, damping_timescale, color="C1", label="Damping time"
+    )
+    top3 = axes[-1].plot(
+        times,
+        smooth_damping_timescale,
+        color="C2",
+        label="Smooth damping time",
+    )
+    top_lines = top1 + top2 + top3
+    top_labels = [l.get_label() for l in top_lines]
+    axes[0].tick_params(axis="y", labelcolor="C0")
+
+    # State, and delta R
+    axes[1].plot(times, state, label="State")
+    axes[1].plot(times, min_delta_r, label=r"Min $\Delta$R")
+    axes[1].plot(times, avg_delta_r, label=r"Average $\Delta$R")
+
+    # Relative delta R
+    axes[2].plot(times, min_relative_delta_r, label=r"Min relative $\Delta$R")
+    axes[2].plot(
+        times, avg_relative_delta_r, label=r"Average relative $\Delta$R"
+    )
+
+    # Char speeds
+    axes[3].plot(times, min_char_speed, label="Min char speed")
+    axes[3].plot(
+        times, min_comoving_char_speed, label="Min comoving char speed"
+    )
+    axes[3].plot(times, target_char_speed, label="Target char speed")
+
+    # Zero crossing predictors
+    axes[4].plot(
+        times, char_speed_crossing_time, label="Char speed crossing time"
+    )
+    axes[4].plot(
+        times,
+        comoving_char_speed_crossing_time,
+        label="Comoving char speed crossing time",
+    )
+    axes[4].plot(
+        times, delta_r_speed_crossing_time, label=r"$\Delta$R crossing time"
+    )
+
+    # Lambda and horizon
+    axes[5].plot(times, function_of_time, label="Function of time")
+    axes[5].plot(times, horizon_coef_00, label="Horizon coef 00")
+
+    # Time deriv of lambda and horizon
+    axes[6].plot(times, dt_function_of_time, label="dtFunction of time")
+    axes[6].plot(times, raw_dt_horizon_coef_00, label="Raw dtHorizon coef 00")
+    axes[6].plot(times, avg_dt_horizon_coef_00, label="Avg dtHorizon coef 00")
+
+    # Configure the axes and legends
+    fig.set_size_inches(10, len(axes) * 2)
+    fig.set_dpi(100)
+    if xlabel:
+        # Bottom plot. -1 is the second y-axis for top plot
+        axes[-2].set_xlabel(xlabel)
+    if xlim:
+        plt.xlim(*xlim)
+    if title:
+        plt.title(title)
+    for i in range(len(axes) - 1):
+        autoscale(axes[i], margin=0.05)
```

Comment:
Can't you just rely on matplotlib's autoscaling? I think a way to also avoid plotting all the data outside the bounds is with `data = data[data["Time"] >= xlim[0] & data["Time"] <= xlim[1]]` right after loading the data.

Reply:
I found that when I limited the x-range, that the y-axes didn't rescale automatically. It kept the autoscaling for the full x-range. However, I agree if we limit the data right after loading, this problem goes away. I'll try that